transcom/mymove
# B 19026 int

## [B-19026](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19026)

## Summary

The goal of this task was to ensure that reviewing an AOA request is required as a service counselor is approving a PPM. It required adding validation and error handling to one group of frontend radio buttons pictured below.

In order to display the proper error messages, the default global radio button type had to be extended. This was completed thanks to help from @cameroncaci who had recently configured [something similar for checkboxes](https://github.com/transcom/mymove/pull/12698).

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the application as a user and create a PPM Shipment with an Advance request.
2. Submit the move.
3. Log in to office app as a Service Counselor.
4. Review the PPM Shipment. On the first page, no changes are necessary.
5. On the second page, it should appear like this:
<img width="1623" alt="image" src="https://github.com/transcom/mymove/assets/148906585/e82b6b1d-3488-4e18-9290-e6ea11b7651b">
6. Ensure you cannot submit the move without approving, denying, or changing the user's request for an AOA.
<img width="1623" alt="image" src="https://github.com/transcom/mymove/assets/148906585/378a80bd-c583-437f-bed0-e517f0880362">
7. Test by editing other information on the shipment, no changes will allow you to bypass this other than approving, denying, or otherwise changing the request for the AOA.
8. Ensure all client and playwright tests are passing.


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
